modules/data-entry/src/main/frontend/src/questionnaire/Subject.jsx

```diff
@@ -223,45 +234,115 @@ function SubjectContainer(props) {
   }
 
   return (
-    data && <Grid container spacing={4} direction="column" className={classes.subjectContainer}>
-      <SubjectMember classes={classes} id={path} level={currentLevel} data={data} maxDisplayed={maxDisplayed} pageSize={pageSize} onDelete={() => {setDeleted(true)}}/>
+    subject && <React.Fragment>
+      <SubjectMember classes={classes} id={id} level={currentLevel} data={subject} maxDisplayed={maxDisplayed} pageSize={pageSize} onDelete={() => {setDeleted(true)}}/>
       {relatedSubjects && relatedSubjects.length > 0 ?
         (<Grid item xs={12} className={classes.subjectNestedContainer}>
           {relatedSubjects.map( (subject, i) => {
             // Render component again for each related subject
             return(
-              <SubjectContainer key={i} classes={classes} path={subject["@path"]} level={currentLevel+1} maxDisplayed={maxDisplayed} pageSize={pageSize}/>
+              <SubjectContainer key={i} classes={classes} path={subject["@path"]} level={currentLevel+1} maxDisplayed={maxDisplayed} pageSize={pageSize} subject={subject}/>
             )
           })}
         </Grid>
         ) : ""
       }
-    </Grid>
+    </React.Fragment>
   );
 }
 
 /**
- * Component that displays the subject chart header, with subject id, parents, and actions
+ * Component that displays the header for the selected subject and its SubjectType
  */
-function SubjectHeader (props) {
-  let {data, classes, title, action} = props;
-  let parentDetails = data && data['parents'] && getHierarchy(data['parents']);
+function SubjectHeader(props) {
+  let { id, classes, getSubject } = props;
+  // This holds the full form JSON, once it is received from the server
+  let [ subject, setSubject ] = useState(null);
+  // Error message set when fetching the data from the server fails
+  let [ error, setError ] = useState();
+  // whether the subject has been deleted
+  let [ deleted, setDeleted ] = useState();
+
+  let globalLoginDisplay = useContext(GlobalLoginContext);
+
+  if (deleted) {
+    return null;
+  }
+
+  // Fetch the subject's data as JSON from the server.
+  // The data will contain the subject metadata,
+  // such as authorship and versioning information.
+  // Once the data arrives from the server, it will be stored in the `data` state variable.
+  let fetchSubjectData = () => {
+    fetchWithReLogin(globalLoginDisplay, `/Subjects/${id}.progeny.deep.json`)
```

Comment:
```suggestion
    fetchWithReLogin(globalLoginDisplay, `/Subjects/${id}.deep.json`)
```
The progeny selector is no longer used as of LFS-911